### ISSUE TITLE
Preventing Build Errors on ph-submissions

### ISSUE THREAD

**Issue by walshbr** [programminghistorian/jekyll#1795]
The broad proposed action here is that we rethink how we ask authors to upload materials to the ph-submissions repository or otherwise think through the editorial workflow slightly differently. 

Ph-submissions has people push directly to it rather than make pull requests (presumably for ease of use by authors). But this means that if the site gets a build error the results are a little more hidden than they would be on our site. You get an email, but the email can be easy to ignore because it won't block the contribution from being part of the repo like it would in a pull request. Here’s an example screenshot of the commit log:

![Screen Shot 2020-05-28 at 2 27 09 PM](https://user-images.githubusercontent.com/2334220/83179772-798b7800-a0f0-11ea-887c-5909e957ef51.png)

There was an issue introduced several commits back that broke the site build, so it stopped rendering properly. The issue with this is that, until the build error is fixed, the build will continue to error until someone fixes it. And it won’t build on the subsequent pushes. That means, for example, that if I had had looked at the live, rendered version of my lesson in progress that I just committed to I might be wondering why changes I pushed weren’t showing up. The errors persist until someone notices them and fixes them, and there’s not really anyone tending to them. Most of the errors eventually get ironed out in the review process, but that can often take a while. They’re primarily errors introduced by authors who have made mistakes with the metadata or include syntax for figures. The build error here actually wasn’t resolved until  made changes to four different lessons in progress. No one is really tending the repo in that way, and there's not a good process for making sure that those sorts of errors don't take place. 

We discussed this a little on the technical team slack how we might avoid this sort of thing in the future. The consensus that I'd like to propose is the following:

- We tweak the editorial process to have the editor for a lesson post the initial file for the lesson and, in the process, make a first pass at making sure the metadata is formatted correctly. 
- We make a note about how the editorial/translation process should involve checking the commit log for build errors and show where to find them in the web interface (I don't think this is in the guidelines, but I might have missed it while re-reading just now). 

That would catch most of the errors I think and give a process for making sure they get caught regularly as part of the process.

**Comment by jenniferisasi** [comment]
I realized that we don't have a tag for "editors only" or similar so this might be missed by them. I am pinging @rivaquiroga @spapastamkou and @svmelton for a quick check as MEs. And maybe we need that tag.

**Comment by walshbr** [comment]
Good point and thanks!

**Comment by rivaquiroga** [comment]
This is a very good idea. And you are right, checking the logs is not on the editor guidelines. Actually, in case the rendered version of the page doesn't work, the propose solution is: "let Matthew Lincoln know, and he will try to diagnose it". This is something we editors should try to solve first.

**Comment by jenniferisasi** [comment]
Oh no, right, that "contact Matt" solution is NOT good at all. In theory, authors/translators should be using the templates with the YAML in place, etc. but...

**Comment by walshbr** [comment]
No wonder he's retiring!

**Comment by mdlincoln** [comment]
>"let Matthew Lincoln know, and he will try to diagnose it"

![](https://media.giphy.com/media/13jCHJXzAJjrDG/giphy.gif)

**Comment by rivaquiroga** [comment]
Some actions arising from this issue:

- Update Author + Editor + Translator guidelines regarding who uploads the initial file for the lesson

- Update Editor Guidelines for checking any build errors after commits from authors/translators

**Comment by spapastamkou** [comment]
Would it be simpler to ask authors/translators to make pull requests reviewed by the team members instead of commiting directly? Or not?

**Comment by walshbr** [comment]
Simpler in terms in one sense, maybe, but it would mean that we might have to give people guidance on how to submit pull requests I think. So the burden on authors becomes a little more complicated.

**Comment by svmelton** [comment]
Just noting that I think this is a good solution—thanks @walshbr!

**Comment by walshbr** [comment]
Seems like there is consensus among the managing editors that this is a good idea, so I'll make these changes to the guidelines.

**Comment by walshbr** [comment]
Just a note that when I start working on this I should

Check with the Portuguese team to make sure that they haven't translated the relevant pages. And, depending on that answer, we'll need to make sure to copy the relevant English page after it's been updated into the Portuguese collection of pages that they're working from.

**Comment by DanielAlvesLABDH** [comment]
> 
> 
> Just a note that when I start working on this I should
> 
> Check with the Portuguese team to make sure that they haven't translated the relevant pages. And, depending on that answer, we'll need to make sure to copy the relevant English page after it's been updated into the Portuguese collection of pages that they're working from.

These are the files already translated to PT, although we haven’t send them to GitHub yet: Index.md, Lessons.md, Project-team.md, Feedback.md, Contribute.md, Privacy-policy.md, About.md, reviewer-guidelines.md and support-us.md. We are in the process of translating Editor-guidelines.md (probably more then half already translated).

**Comment by walshbr** [comment]
Thanks @DanielAlvesLABDH - in that case I will let you know when I have new text on the Editor Guidelines and Author guidelines and ping you along with the other translation teams. It will be some time until I can get to it anyway, and I don't want to hang you up. The translation shouldn't be super burdensome - I need to look at the text to be certain, but it certainly won't be mass re-translating. Quickly glancing, it looks like there will 5-6 affected sentences in the author-guidelines.md document (under "submitting a lesson") and, in the editor-guidlines, a small revision to the "track proposed lesson" section.

**Comment by walshbr** [comment]
Just a note here that @jenniferisasi noticed a related issue while the tech team was discussing this issue. The ph-submissions repo is a bit of a mess - we have en and fr folders with lessons inside of them, but everything for en goes inside a lessons folder instead, and that same lessons folder contains different fr, pt, and es folders. So the structure is pretty confusing and disordered. We came up with a solution that we're going to try in conversation with @ZoeLeBlanc. I emailed the managing editors first, but we'll update #1844 with the fix when it's cleared. Note that the fix will apply things for **future lessons** only - we'll let the old lessons uploaded in weird places under the old system finish and close out rather than trying to move a ton of stuff around. So it should be the case that this will clean up the future and then past will quietly move along. 

- Update: it actually makes more sense to break this off as its own task to avoid lumping too much into one PR. I'll open a separate ticket to that effect.
